Golem/Jobs.cs

```diff
@@ -193,18 +205,31 @@ public async Task UpdatePartialPayment(Payment payment)
         }
     }
 
-    private async Task<Job> UpdateJobStatus(string agreementId)
+    public async Task<Job> UpdateJobStatus(string agreementId)
     {
         var job = await GetOrCreateJob(agreementId);
         var agreement = await _yagna.Api.GetAgreement(agreementId);
 
         // Agreement state has precedens over individual activity states.
         if (agreement.State == "Terminated")
         {
-            job.Status = JobStatus.Finished;
+            var termination = await _yagna.Api.GetTerminationReason(agreementId);
```

Comment:
if the agreement is terminated then this call throws an Exception and ends up polluting the logs/events with lots of similar error messages:

![image](https://github.com/golemfactory/gamerhash-facade/assets/16387248/3f7f1b4e-61c2-4647-93aa-0c63f91e94a7)